internal/dag/status_test.go

```diff
@@ -3356,6 +3356,251 @@ func TestDAGStatus(t *testing.T) {
+			Routes: []contour_api_v1.Route{
+				{
+					JWTVerificationPolicy: &contour_api_v1.JWTVerificationPolicy{
+						Require: "provider-1",
+					},
+					Conditions: []contour_api_v1.MatchCondition{{
+						Prefix: "/",
+					}},
+					Services: []contour_api_v1.Service{{
+						Name: "home",
+						Port: 8080,
+					}},
+				},
+			},
+		},
+	}
+
+	run(t, "JWT verification invalid upstream validation specified for HTTP JWKS", testcase{
+		objs: []interface{}{
+			jwtVerificationUpstreamValidationForHTTPJWKS,
+			fixture.SecretRootsCert,
+			fixture.ServiceRootsHome,
+		},
+		want: map[types.NamespacedName]contour_api_v1.DetailedCondition{
+			k8s.NamespacedNameOf(jwtVerificationUpstreamValidationForHTTPJWKS): fixture.NewValidCondition().
+				WithError(
+					contour_api_v1.ConditionTypeJWTVerificationError,
+					"RemoteJWKSUpstreamValidationInvalid",
+					"Spec.VirtualHost.JWTProviders.RemoteJWKS.UpstreamValidation must not be specified when URI scheme is http.",
+				),
+		},
+	})
+
+	jwtVerificationUpstreamValidationCACertDoesNotExist := &contour_api_v1.HTTPProxy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "roots",
+			Name:      "jwt-verification-upstream-validation-cacert-does-not-exist",
+		},
+		Spec: contour_api_v1.HTTPProxySpec{
+			VirtualHost: &contour_api_v1.VirtualHost{
+				Fqdn: "example.com",
+				TLS: &contour_api_v1.TLS{
+					SecretName: fixture.SecretRootsCert.Name,
+				},
+				JWTProviders: []contour_api_v1.JWTProvider{
+					{
+						Name: "provider-1",
+						RemoteJWKS: contour_api_v1.RemoteJWKS{
+							URI: "https://jwt.example.com/jwks.json",
+							UpstreamValidation: &contour_api_v1.UpstreamValidation{
+								CACertificate: "nonexistent",
+								SubjectName:   "jwt.example.com",
+							},
+						},
+					},
+				},
+			},
+			Routes: []contour_api_v1.Route{
+				{
+					JWTVerificationPolicy: &contour_api_v1.JWTVerificationPolicy{
+						Require: "provider-1",
+					},
+					Conditions: []contour_api_v1.MatchCondition{{
+						Prefix: "/",
+					}},
+					Services: []contour_api_v1.Service{{
+						Name: "home",
+						Port: 8080,
+					}},
+				},
+			},
+		},
+	}
+
+	run(t, "JWT verification invalid upstream validation CA cert does not exist", testcase{
+		objs: []interface{}{
+			jwtVerificationUpstreamValidationCACertDoesNotExist,
+			fixture.SecretRootsCert,
+			fixture.ServiceRootsHome,
+		},
+		want: map[types.NamespacedName]contour_api_v1.DetailedCondition{
+			k8s.NamespacedNameOf(jwtVerificationUpstreamValidationCACertDoesNotExist): fixture.NewValidCondition().
+				WithError(
+					contour_api_v1.ConditionTypeJWTVerificationError,
+					"RemoteJWKSUpstreamValidationInvalid",
+					"Spec.VirtualHost.JWTProviders.RemoteJWKS.UpstreamValidation is invalid: invalid CA Secret \"roots/nonexistent\": Secret not found",
+				),
+		},
+	})
+
+	jwksInvalidCACert := &v1.Secret{
+		ObjectMeta: fixture.ObjectMeta("roots/cacert"),
+		Data: map[string][]byte{
+			"wrong-key": []byte(fixture.CERTIFICATE),
```

Comment:
i guess this case causes the secret to never be added to the cache, which results in a not found error similar to the above? I'm not sure if we can even test something distinct to that, since we do so much checking of secret validity when adding to the cache

the check for a valid CA when looking it up from the cache is almost moot it seems

might be ok to remove this test case

Reply:
Yep, I realized the same when I implemented this test case. I figured we may as well leave it so it's covered in case of future changes in this area.